rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java

```diff
@@ -1135,4 +1137,77 @@ void mergeEmptyStructureFollowedByCopyValue() {
           )
         );
     }
+
+    @Test
+    void updateOnly() {
+        rewriteRun(
+          spec -> spec
+            .recipeFromYaml(
+              //language=yaml
+              """
+                ---
+                type: specs.openrewrite.org/v1beta/recipe
+                name: org.openrewrite.Update
+                displayName: Update only
+                description: Update a yaml file.
+                recipeList:
+                  - org.openrewrite.yaml.MergeYaml:
+                        key: $.some.object
+                        yaml: |
+                          other: A new value
+                ""","org.openrewrite.Update"),
+          yaml(
+            //language=yaml
+            """
+            some:
+              object:
+                with: An existing value
+            """,
+            //language=yaml
+            """
+              some:
+                object:
+                  with: An existing value
+                  other: A new value
+              """, spec -> spec.path(Path.of("updated.yml")))
+        );
+    }
+
+    @Test
+    void createAndUpdate() {
+        rewriteRun(
+          spec -> spec
+            .recipeFromYaml(
+              //language=yaml
+              """
+                ---
+                type: specs.openrewrite.org/v1beta/recipe
+                name: org.openrewrite.CreateAndUpdate
+                displayName: Create and update
+                description: Create a file and update it.
+                recipeList:
+                  - org.openrewrite.text.CreateTextFile:
```

Comment:
Not sure where this example came from, but we do have an explicit Create YAML file recipe: `org.openrewrite.yaml.CreateYamlFile`
https://docs.openrewrite.org/recipes/yaml/createyamlfile

Might be worth either passing along that message, or testing with a Yaml file created.

Reply:
yes, I found that, an ran into the next issue 🙈